File: ruby/bob/bob.rb
class Bob
  def hey remark
    return 'Fine. Be that way!' if remark.match(/^\s+$/) || remark==''
    return 'Whoa, chill out!' if remark==remark.upcase && !remark.match(/^[\d, \?]+$/)
    return 'Sure.' if remark[-1]=='?'
    'Whatever.'
  end
end

File: python/bob/bob.py
import re

def hey(what):
	if re.match(r'^\s+$', what) or what == "":
		return 'Fine. Be that way!'
	if what == what.upper() and not re.match(r'^[\d, \?]+$', what):
		return 'Whoa, chill out!'
	if what.strip()[-1:] == '?':
		return 'Sure.'
	return 'Whatever.'
